homeassistant/components/modem_callerid/sensor.py

```diff
@@ -84,38 +101,53 @@ def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the sensor."""
+        return f"{self._server_unique_id}/{self._name}"
```

Comment:
Name doesn't meet the unique id requirements. 
```suggestion
        return f"{self._server_unique_id}_modem"
```

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements